.github/actions/setup-binary/action.yml

```diff
@@ -0,0 +1,117 @@
+# Copyright 2023 The Authors (see AUTHORS file)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: 'Setup Binary'
+
+description: |-
+  Use this action to install and cache a binary for use within a GitHub workflow.
+
+inputs:
+  download_url:
+    description: 'The download URL for the binary. This must be of the format https://...'
+    required: true
+  install_path:
```

Comment:
Did you mean to get rid of the feature that let you specify the path _within_ the archive for the binary?

Reply:
Yes, what is it used for? If we extract everything, the user can specify the sub path when assigning to the $PATH.

Reply:
I wonder if it would be simpler if we default to adding the root of the downloaded directory (in case of tarball/zip) to the PATH or the path to the binary (in case of no extension). Then we can provide an optional input for when the tarball has a nested binary. But the default use case is simply - 
```
uses: "setup-binary/action.yml"
with:
  download_url: "https://dasdasda/asdasd..."
```

Reply:
Yea I guess if we extract everything, it's up to the end-user to know if the intended binary is in a sub-folder.

I don't think it's a good idea to modify the PATH without user consent. So we can either:

1. Put the binary in a well-known location (e.g. /bin)
2. Accept a `add_to_path` input that defaults to false
3. Expose an output that provides the extracted archives path and provide an example for adding that value to $PATH

Reply:
I would prefer 2 or 3

For 2, we would then need to bring back the subpath input for creating the path?

For 3, do we need that? They gave us the install_path, that should be the same no?